drake/solvers/MathematicalProgram.cpp

```diff
@@ -31,41 +31,41 @@ class MathematicalProgram : public MathematicalProgramInterface {
    virtual MathematicalProgramInterface* addComplementarityConstraint() { return new
    MathematicalProgram; };
 */
-  virtual MathematicalProgramInterface* addGenericObjective() override {
+  virtual MathematicalProgramInterface* add_generic_objective() override {
```

Comment:
All operations on MathematicalProgramInterface are just using the type system for lattice algebra; each one simply allocates a vacuous object, and is equivalently costly to a malloc.